diff --git a/composer.json b/composer.json
index 0a8451ee70c63c600bb9903e23d14c5eb9834a5a..a7f0ea2d42df85bd7ccab38df25a283649383fe5 100644
--- a/composer.json
+++ b/composer.json
@@ -11,7 +11,7 @@
         "source": "http://cgit.drupalcode.org/social_auth_hid"
     },
     "require": {
-        "un-ocha/oauth2-hid": "^1.0.1"
+        "un-ocha/oauth2-hid": "^1.0.2"
     },
     "authors": [
         {
diff --git a/config/install/social_auth_hid.settings.yml b/config/install/social_auth_hid.settings.yml
new file mode 100644
index 0000000000000000000000000000000000000000..881d2c936e29aa5f3372eb1ec5e45c095762ed12
--- /dev/null
+++ b/config/install/social_auth_hid.settings.yml
@@ -0,0 +1 @@
+base_url: 'https://auth.humanitarian.id'
diff --git a/config/schema/social_auth_hid.schema.yml b/config/schema/social_auth_hid.schema.yml
index 04cc3c942ef501ee07a83dd746bcee384cc5a9ae..ee5105bfc67470d49776183395a5557b01063107 100644
--- a/config/schema/social_auth_hid.schema.yml
+++ b/config/schema/social_auth_hid.schema.yml
@@ -8,3 +8,6 @@ social_auth_hid.settings:
     client_secret:
       type: string
       label: 'Client Secret'
+    base_url:
+      type: string
+      label: 'HID base URL'
diff --git a/social_auth_hid.post_update.php b/social_auth_hid.post_update.php
new file mode 100644
index 0000000000000000000000000000000000000000..97f2de1dd6cee5a0cef3939065d2d09d7d7ebee3
--- /dev/null
+++ b/social_auth_hid.post_update.php
@@ -0,0 +1,17 @@
+<?php
+
+/**
+ * @file
+ * Post update functions for the Social Auth HID module.
+ */
+
+/**
+ * Adds HID base URL setting.
+ */
+function social_auth_hid_add_url_setting() {
+  $config = \Drupal::configFactory()
+    ->getEditable('social_auth_hid.settings');
+
+  $config->set('base_url', 'https://auth.humanitarian.id')
+    ->save(TRUE);
+}
diff --git a/src/Form/HidAuthSettingsForm.php b/src/Form/HidAuthSettingsForm.php
index fa89bf4571645dda3d200a7ffa3a686b82aeb91e..fdb0e7af1d2c136be81769fe9f70421520de429e 100644
--- a/src/Form/HidAuthSettingsForm.php
+++ b/src/Form/HidAuthSettingsForm.php
@@ -3,6 +3,7 @@
 namespace Drupal\social_auth_hid\Form;
 
 use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Component\Utility\UrlHelper;
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\Core\Path\PathValidatorInterface;
 use Drupal\Core\Routing\RequestContext;
@@ -99,9 +100,35 @@ class HidAuthSettingsForm extends SocialAuthSettingsForm {
       '#description' => $this->t('Copy the Client Secret here.'),
     ];
 
+    $form['hid_settings']['advanced'] = [
+      '#type' => 'details',
+      '#title' => $this->t('Advanced settings'),
+      '#open' => FALSE,
+    ];
+
+    $form['hid_settings']['advanced']['base_url'] = [
+      '#type' => 'textfield',
+      '#required' => TRUE,
+      '#title' => $this->t('HID base URL'),
+      '#description' => $this->t('Customize your HID base URL, e.g. if you want to use HID dev or HID staging.'),
+      '#default_value' => $config->get('base_url'),
+    ];
+
+
     return parent::buildForm($form, $form_state);
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    $values = $form_state->getValues();
+    parent::validateForm($form, $form_state);
+    if (!UrlHelper::isValid($values['base_url'], TRUE)) {
+      $form_state->setErrorByName('base_url', $this->t("The HID base URL is invalid."));
+    }
+  }
+
   /**
    * {@inheritdoc}
    */
@@ -110,6 +137,7 @@ class HidAuthSettingsForm extends SocialAuthSettingsForm {
     $this->config('social_auth_hid.settings')
       ->set('client_id', trim($values['client_id']))
       ->set('client_secret', trim($values['client_secret']))
+      ->set('base_url', rtrim($values['base_url'], '/'))
       ->save();
 
     parent::submitForm($form, $form_state);
diff --git a/src/HidAuthManager.php b/src/HidAuthManager.php
index 630ced840d9fd07cdc4a0e1ccad663ec7f17de41..cefdb3d47260a2682991f395e14d755ca51014b6 100644
--- a/src/HidAuthManager.php
+++ b/src/HidAuthManager.php
@@ -68,7 +68,7 @@ class HidAuthManager extends OAuth2Manager {
    */
   public function requestEndPoint($method, $path, $domain = NULL, array $options = Array()) {
     if (!$domain) {
-      $domain = 'https://auth.humanitarian.id';
+      $domain = $this->settings->get('base_url');
     }
     $url = $domain . $path;
 
diff --git a/src/Plugin/Network/HidAuth.php b/src/Plugin/Network/HidAuth.php
index 2a421b15a989822c8f175f66d3da90d1bdff2f36..2635b8950e11e32a0eb58716b6cfcab5fc26bd2b 100644
--- a/src/Plugin/Network/HidAuth.php
+++ b/src/Plugin/Network/HidAuth.php
@@ -144,6 +144,7 @@ class HidAuth extends NetworkBase implements HidAuthInterface {
         'clientId' => $settings->getClientId(),
         'clientSecret' => $settings->getClientSecret(),
         'redirectUri' => $redirect_url,
+        'domain' => $settings->getBaseUrl(),
       ];
 
       // Proxy configuration data for outward proxy.
diff --git a/src/Settings/HidAuthSettings.php b/src/Settings/HidAuthSettings.php
index ef340caf899d8e35c0ecc4ffcdf93691eb5f4abd..2ee1def6daaa36d73db029dac901bed9dbd2d270 100644
--- a/src/Settings/HidAuthSettings.php
+++ b/src/Settings/HidAuthSettings.php
@@ -23,6 +23,13 @@ class HidAuthSettings extends SettingsBase implements HidAuthSettingsInterface {
    */
   protected $clientSecret;
 
+  /**
+   * The base URL.
+   *
+   * @var string
+   */
+  protected $baseUrl;
+
   /**
    * {@inheritdoc}
    */
@@ -43,4 +50,14 @@ class HidAuthSettings extends SettingsBase implements HidAuthSettingsInterface {
     return $this->clientSecret;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function getBaseUrl() {
+    if (!$this->baseUrl) {
+      $this->baseUrl = $this->config->get('base_url');
+    }
+    return $this->baseUrl;
+  }
+
 }
diff --git a/src/Settings/HidAuthSettingsInterface.php b/src/Settings/HidAuthSettingsInterface.php
index ddbf3876ba17e59e384e95aac981848e77469fe7..d4ab5e106543b579d70ca09bf17446296720ad77 100644
--- a/src/Settings/HidAuthSettingsInterface.php
+++ b/src/Settings/HidAuthSettingsInterface.php
@@ -23,4 +23,12 @@ interface HidAuthSettingsInterface {
    */
   public function getClientSecret();
 
+  /**
+   * Gets the base URL.
+   *
+   * @return string
+   *   The HID base URL.
+   */
+  public function getBaseUrl();
+
 }
